src/common/libkvs/jansson_dirent.c

```diff
@@ -0,0 +1,127 @@
+/*****************************************************************************\
+ *  Copyright (c) 2014 Lawrence Livermore National Security, LLC.  Produced at
+ *  the Lawrence Livermore National Laboratory (cf, AUTHORS, DISCLAIMER.LLNS).
+ *  LLNL-CODE-658032 All rights reserved.
+ *
+ *  This file is part of the Flux resource manager framework.
+ *  For details, see https://github.com/flux-framework.
+ *
+ *  This program is free software; you can redistribute it and/or modify it
+ *  under the terms of the GNU General Public License as published by the Free
+ *  Software Foundation; either version 2 of the license, or (at your option)
+ *  any later version.
+ *
+ *  Flux is distributed in the hope that it will be useful, but WITHOUT
+ *  ANY WARRANTY; without even the IMPLIED WARRANTY OF MERCHANTABILITY or
+ *  FITNESS FOR A PARTICULAR PURPOSE.  See the terms and conditions of the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.
+ *  See also:  http://www.gnu.org/licenses/
+\*****************************************************************************/
+
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+#include <string.h>
+#include <assert.h>
+#include <errno.h>
+#include <jansson.h>
+
+#include "src/common/libutil/blobref.h"
+#include "src/common/libutil/oom.h"
+
+#include "jansson_dirent.h"
+
+json_t *j_dirent_create (const char *type, void *arg)
+{
+    json_t *dirent;
+    bool valid_type = false;
+
+    if (!(dirent = json_object ()))
+        oom ();
+
+    if (!strcmp (type, "FILEREF") || !strcmp (type, "DIRREF")) {
+        char *ref = arg;
+        json_t *o;
+
+        if (!(o = json_string (ref)))
+            oom ();
+        if (json_object_set_new (dirent, type, o) < 0)
+            oom ();
+
```

Comment:
Perhaps @garlick and I talked about this offline.  The "classic" dirent functions (```dirent_create()```, etc.) all used ```oom()``` behind the scenes.  I think refactoring the jansson equivalents (```j_dirent_create()```, etc.) to return ENOMEM and all callers to handle this is probably a bit much refactoring for this round.   Figured to leave the ```oom()``` behavior and leave that refactoring for another round.